### PR TITLE
feat(testing): allow omitting `httpAdapter` arg on `createNestApplication`

### PR DESCRIPTION
**Pull request by micalevisk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

When using `createNestApplication` from `Test.createTestingModule()`, the only way to pass options while still using the default http adapter is by passing `undefined` (or any falsy value) as the first arg like this:

```ts
moduleFixture = await Test.createTestingModule({}).compile()
app = moduleFixture.createNestApplication(undefined, { bodyParser: false })
```

but to me this interface doesn't feels good from DX POV. See https://github.com/nestjs/docs.nestjs.com/issues/2119 (which can be closed if we merge this PR)

## What is the new behavior?

the options can be supplied in the first arg if we want to use the default http adatper, like so:

```ts
app = moduleFixture.createNestApplication({ bodyParser: false }) // THIS IS NEW

// Same as:
app = moduleFixture.createNestApplication(undefined, { bodyParser: false }) // v1
app = moduleFixture.createNestApplication(new ExpressAdapter(), { bodyParser: false }) // v2
```

_I tested both locally_

which I believe is more intuitve than passing `undefined`.

Or maybe will be better if we drop `v1` and move this to nestjs v9 (as it introduces breaking changes)? :thinking: 

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/testing/testing-module.ts [hunk 1]
```diff
@@ -28,11 +28,33 @@ export class TestingModule extends NestApplicationContext {
...
   }
 
+  private isHttpServer(
+    httpAdapterOrOptions?:
+      | HttpServer
+      | AbstractHttpAdapter
+      | NestApplicationOptions,
+  ): httpAdapterOrOptions is HttpServer | AbstractHttpAdapter {
+    return !!(
+      httpAdapterOrOptions && (httpAdapterOrOptions as HttpServer).patch
```
Comment (micalevisk) [inline packages/testing/testing-module.ts:38]: I copy this logic from 

https://github.com/nestjs/nest/blob/6f5986d19e43da92801ac3f7d59cab9554741550/packages/core/nest-factory.ts#L257-L263

File: packages/testing/testing-module.ts [hunk 2]
```diff
@@ -28,20 +28,35 @@ export class TestingModule extends NestApplicationContext {
     super(container, scope, contextModule);
   }
 
+  private isHttpServer(
+    serverOrOptions?: HttpServer | AbstractHttpAdapter | NestApplicationOptions,
+  ): serverOrOptions is HttpServer | AbstractHttpAdapter {
+    return !!(serverOrOptions && (serverOrOptions as HttpServer).patch);
```
Comment (micalevisk) [inline packages/testing/testing-module.ts:34]: I copy this logic from 

https://github.com/nestjs/nest/blob/6f5986d19e43da92801ac3f7d59cab9554741550/packages/core/nest-factory.ts#L257-L263

Idk why we're allowing `HttpServer` here tho, as `AbstractHttpAdapter` implements `HttpServer`

**Comment by kamilmysliwiec** [comment]
Let's release this as part of v9. Can you rebase @micalevisk?

**Comment by micalevisk** [comment]
I'm struggling on making this rebase lol I'll just open another PR instead

**Pull request closed without merging**